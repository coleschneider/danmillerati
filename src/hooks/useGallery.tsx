import * as React from "react";
import images, { imageFetchStatus } from "../pages/Gallery/imagePaths";

const imageStatusReducer = (state: ImageStatus, action: GalleryActions) => {
    switch (action.type) {
        case "IMAGE_LOADED":
            return {
                ...state,
                [action.payload.name]: true
            };
        default:
            return state;
    }
};

const lightboxReducer = (state: Lightbox, { type }: GalleryActions) => {
    switch (type) {
        case "OPEN_LIGHTBOX":
            return {
                ...state,
                isOpen: true
            };
        case "CLOSE_LIGHTBOX":
            return {
                ...state,
                isOpen: false
            };
        case "OPEN_FULLSCREEN":
            return {
                ...state,
                isFullscreen: true
            };
        case "CLOSE_FULLSCREEN":
            return {
                ...state,
                isFullscreen: false
            };
        default:
            return state;
    }
};
const carouselReducer = (state: Carousel, action: GalleryActions) => {
    switch (action.type) {
        case "SET_IMAGE":
            return action.payload.activeImage;
        case "NEXT_IMAGE":
            if (state === images.length - 1) {
                return 0;
            }
            return state + 1;
        case "PREV_IMAGE":
            if (state === 0) {
                return images.length - 1;
            }
            return state - 1;
        default:
            return state;
    }
};

const initialState: GalleryState = {
    carousel: 0,
    imageStatus: imageFetchStatus,
    lightbox: {
        isOpen: false,
        isFullscreen: false
    }
};
const galleryReducer = (
    { carousel, imageStatus, lightbox }: GalleryState,
    action: GalleryActions
): GalleryState => ({
    carousel: carouselReducer(carousel, action),
    imageStatus: imageStatusReducer(imageStatus, action),
    lightbox: lightboxReducer(lightbox, action)
});

function useGallery(initState: GalleryState = initialState): UseGallery {
    const [state, dispatch] = React.useReducer(galleryReducer, initState);
    const openLightbox = () =>
        dispatch({
            type: "OPEN_LIGHTBOX"
        });
    const closeLightbox = () =>
        dispatch({
            type: "CLOSE_LIGHTBOX"
        });
    const openFullscreen = () =>
        dispatch({
            type: "OPEN_FULLSCREEN"
        });
    const closeFullscreen = () =>
        dispatch({
            type: "CLOSE_FULLSCREEN"
        });
    const loadImage = (name: ImageName) =>
        dispatch({
            type: "IMAGE_LOADED",
            payload: {
                name
            }
        });
    const setImage = (activeImage: number) =>
        dispatch({
            type: "SET_IMAGE",
            payload: {
                activeImage
            }
        });
    const goNextImage = () =>
        dispatch({
            type: "NEXT_IMAGE"
        });
    const goPrevImage = () =>
        dispatch({
            type: "PREV_IMAGE"
        });

    return {
        openLightbox,
        closeLightbox,
        openFullscreen,
        closeFullscreen,
        loadImage,
        setImage,
        goNextImage,
        goPrevImage,
        ...state
    };
}

export default useGallery;
