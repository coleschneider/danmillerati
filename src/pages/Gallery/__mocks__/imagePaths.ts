const imagePaths: GalleryImage[] = [
    {
        name: "Backtobacklounge.jpg",
        height: 190,
        width: 190,
        description: "",
        isLoading: false,
        isCached: true,
        // eslint-disable-next-line
        src: require("../../../photos/Backtobacklounge.jpg")
    },
    {
        name: "electricpontoon.jpg",
        height: 190,
        width: 190,
        description: "",
        isLoading: false,
        isCached: true,
        // eslint-disable-next-line
        src: require("../../../photos/electricpontoon.jpg")
    },
    {
        name: "32roadsterred.jpg",
        height: 384,
        width: 576,
        description: "",
        isLoading: false,
        isCached: false,
        // eslint-disable-next-line
        src: require("../../../photos/32roadsterred.jpg")
    },
    {
        name: "356Ccabdoor.jpg",
        height: 216,
        width: 288,
        description: "",
        isLoading: false,
        isCached: false,
        // eslint-disable-next-line
        src: require("../../../photos/356Ccabdoor.jpg")
    }
];

export const imageState = {
    "Backtobacklounge.jpg": {
        name: "Backtobacklounge.jpg",
        height: 190,
        width: 190,
        description: "",
        isLoading: false,
        isCached: true,
        // eslint-disable-next-line
        src: require("../../../photos/Backtobacklounge.jpg")
    },
    "electricpontoon.jpg": {
        name: "electricpontoon.jpg",
        height: 190,
        width: 190,
        description: "",
        isLoading: false,
        isCached: true,
        // eslint-disable-next-line
        src: require("../../../photos/electricpontoon.jpg")
    },
    "32roadsterred.jpg": {
        name: "32roadsterred.jpg",
        height: 384,
        width: 576,
        description: "",
        isLoading: false,
        isCached: false,
        // eslint-disable-next-line
        src: require("../../../photos/32roadsterred.jpg")
    },
    "356Ccabdoor.jpg": {
        name: "356Ccabdoor.jpg",
        height: 216,
        width: 288,
        description: "",
        isLoading: false,
        isCached: false,
        // eslint-disable-next-line
        src: require("../../../photos/356Ccabdoor.jpg")
    }
};
export default imagePaths;
