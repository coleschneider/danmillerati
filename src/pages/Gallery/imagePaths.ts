const images: ImageProperties[] = [
    { name: "Backtobacklounge.jpg", height: 190, width: 190, description: "" },
    { name: "electricpontoon.jpg", height: 190, width: 190, description: "" },
    { name: "32roadsterred.jpg", height: 384, width: 576, description: "" },
    { name: "356Ccabdoor.jpg", height: 216, width: 288, description: "" },
    {
        name: "356backrestrebuildkit.jpg",
        height: 300,
        width: 240,
        description: ""
    },
    { name: "356cCOUPEREARwall.jpg", height: 216, width: 288, description: "" },
    { name: "356cabboot.jpg", height: 301, width: 500, description: "" },
    {
        name: "1968-red-mustang-top-up.jpg",
        height: 470,
        width: 800,
        description: ""
    },
    {
        name: "32interiorredroadster.jpg",
        height: 1152,
        width: 1728,
        description: ""
    },
    { name: "33jimdavisINT.jpg", height: 1855, width: 1371, description: "" },
    { name: "33jimdaviscar.jpg", height: 718, width: 1814, description: "" },
    { name: "356coupeinterior.jpg", height: 360, width: 576, description: "" },
    {
        name: "1965FordMustangConvertibleSide.jpg",
        height: 475,
        width: 799,
        description: ""
    },
    {
        name: "1Motorcycleseatrestoration.jpg",
        height: 1105,
        width: 1233,
        description: ""
    },
    { name: "jagthumb.jpg", height: 206, width: 350, description: "" },
    { name: "32interior.jpg", height: 683, width: 1029, description: "" },
    { name: "carseatsonly.jpg", height: 275, width: 282, description: "" },
    { name: "356speedster.jpg", height: 871, width: 1292, description: "" },
    { name: "1980honda750K.jpg", height: 820, width: 1222, description: "" },
    { name: "32coupeFlkr.jpg", height: 805, width: 1419, description: "" },
    { name: "1978HondaSS.jpg", height: 1152, width: 1728, description: "" },
    { name: "39fordcoupeHULL.jpg", height: 964, width: 2521, description: "" },
    {
        name: "Mustangtopscompilation.jpg",
        height: 1441,
        width: 2529,
        description: ""
    },
    { name: "bm32Frt.jpg", height: 667, width: 1000, description: "" },
    { name: "sprinter1.jpg", height: 1152, width: 1728, description: "" },
    {
        name: "porscheturbointerior_4640.jpg",
        height: 1152,
        width: 1728,
        description: ""
    },
    { name: "53BuickSuper.jpg", height: 2276, width: 2308, description: "" },
    { name: "BillsEconolinevan.jpg", height: 640, width: 640, description: "" },
    { name: "IMG_4021.jpg", height: 1152, width: 1728, description: "" },
    {
        name: "32roadstertomhaynes2.jpg",
        height: 1352,
        width: 1883,
        description: ""
    },
    { name: "bobberseat.jpg", height: 1152, width: 1728, description: "" },
    { name: "Nelsonnija2.jpg", height: 1152, width: 1728, description: "" },
    {
        name: "porscheturbosmaller.jpg",
        height: 1379,
        width: 1726,
        description: ""
    },
    { name: "Blackiesdash.jpg", height: 1152, width: 1728, description: "" },
    { name: "BM32roadster.jpg", height: 819, width: 1126, description: "" },
    { name: "lid.jpg", height: 1152, width: 1728, description: "" },
    { name: "DMATIHEADER.jpg", height: 764, width: 1354, description: "" },
    { name: "kens58big.jpg", height: 1152, width: 1728, description: "" },
    { name: "rollsdash_3200.jpg", height: 1936, width: 2592, description: "" },
    { name: "62RRbacktrays.jpg", height: 598, width: 800, description: "" },
    { name: "blownhemi32.jpg", height: 819, width: 1126, description: "" },
    { name: "truckseatbuttons.jpg", height: 768, width: 1024, description: "" },
    { name: "Olds442conv.jpg", height: 734, width: 2137, description: "" },
    { name: "rollsdoors_3199.jpg", height: 1936, width: 2592, description: "" },
    { name: "DiamondHD.jpg", height: 6430, width: 8942, description: "" },
    { name: "Rightfrontcornner.jpg", height: 366, width: 657, description: "" },
    { name: "woody_3331.jpg", height: 1936, width: 2592, description: "" },
    { name: "rod_n_wheels_a_l.jpg", height: 318, width: 565, description: "" },
    { name: "RobbinsLogo.jpg", height: 322, width: 600, description: "" },
    { name: "GT500Conv.jpg", height: 628, width: 2096, description: "" },
    { name: "62RR.jpg", height: 598, width: 800, description: "" },
    {
        name: "67Chevellesmallr.jpg",
        height: 1124,
        width: 1094,
        description: ""
    },
    { name: "frontF1.jpg", height: 346, width: 616, description: "" },
    { name: "tommyotisraodster.jpg", height: 384, width: 576, description: "" },
    { name: "58exterior.jpg", height: 1152, width: 1728, description: "" },
    { name: "cobraorange2.jpg", height: 1458, width: 2592, description: "" },
    { name: "Petesgalax.jpg", height: 392, width: 629, description: "" },
    { name: "69novaside.jpg", height: 831, width: 1868, description: "" },
    {
        name: "Ferrariinterio3r.jpg",
        height: 2241,
        width: 3000,
        description: ""
    },
    { name: "46fordconv.jpg", height: 740, width: 1017, description: "" },
    { name: "rolls81_2880.jpg", height: 1936, width: 2592, description: "" },
    { name: "NelsenNinja.jpg", height: 1152, width: 1728, description: "" },
    { name: "IMG_4015.jpg", height: 1152, width: 1728, description: "" },
    { name: "rollsfront_3198.jpg", height: 1936, width: 2592, description: "" },
    { name: "woodyreararea.jpg", height: 1936, width: 2592, description: "" },
    { name: "rollsback_3187.jpg", height: 1936, width: 2592, description: "" },
    { name: "cobraorange.jpg", height: 918, width: 1632, description: "" },
    {
        name: "heartbeatHDlowres.jpg",
        height: 1666,
        width: 2231,
        description: ""
    },
    { name: "sidef1.jpg", height: 485, width: 810, description: "" },
    { name: "IMG_2096.jpg", height: 1936, width: 2592, description: "" },
    {
        name: "VWlatemodelwhiteSeatsfront.jpg",
        height: 864,
        width: 576,
        description: ""
    },
    {
        name: "Customchopper3049.jpg",
        height: 1936,
        width: 2592,
        description: ""
    },
    {
        name: "porscheturbostrasek_4644.jpg",
        height: 1152,
        width: 1728,
        description: ""
    },
    { name: "IMG_7407.jpg", height: 1152, width: 1728, description: "" },
    { name: "trnkbox.jpg", height: 1152, width: 1728, description: "" },
    { name: "IMG_2097.jpg", height: 1936, width: 2592, description: "" },
    { name: "miata.jpg", height: 1740, width: 2632, description: "" },
    {
        name: "67ChevelleitalianleatherInterior.jpg",
        height: 3195,
        width: 2365,
        description: ""
    },
    { name: "sprinter3.jpg", height: 1728, width: 1152, description: "" },
    { name: "IMG_4019.jpg", height: 1152, width: 1728, description: "" },
    { name: "Bugeye.jpg", height: 373, width: 734, description: "" },
    { name: "Hondas2000.jpg", height: 179, width: 282, description: "" },
    {
        name: "Jauge-Jense-Viking-Concept-Bike-by-Henrik-Fisker-Seat.jpg",
        height: 959,
        width: 1280,
        description: ""
    },
    { name: "int1.jpg", height: 1152, width: 1728, description: "" },
    { name: "bobberseat2.jpg", height: 1152, width: 1728, description: "" },
    {
        name: "Customchopper_3047.jpg",
        height: 1936,
        width: 2592,
        description: ""
    },
    {
        name: "Hollywoodcarclub.jpg",
        height: 1329,
        width: 1031,
        description: ""
    },
    {
        name: "VintagePorscheSpecialist.jpg",
        height: 1553,
        width: 2265,
        description: ""
    },
    { name: "IMG_7397.jpg", height: 1152, width: 1728, description: "" },
    { name: "Woodyrear_3337.jpg", height: 1936, width: 2592, description: "" },
    { name: "58interior.jpg", height: 1152, width: 1728, description: "" },
    {
        name: "Smallskiboatseatsjpg.jpg",
        height: 275,
        width: 400,
        description: ""
    },
    {
        name: "Jauge-Jense-Viking-Concept-Bike-by-Henrik-Fisker-05.jpg",
        height: 851,
        width: 1280,
        description: ""
    },
    { name: "blackies33.jpg", height: 1152, width: 1728, description: "" },
    { name: "63vwrightside.jpg", height: 960, width: 1280, description: "" },
    { name: "IMG_0908.jpg", height: 426, width: 640, description: "" },
    { name: "Bill49.jpg", height: 376, width: 640, description: "" },
    { name: "porscheturbo.jpg", height: 2186, width: 2736, description: "" },
    { name: "Bugeyetrunk.jpg", height: 471, width: 730, description: "" },
    { name: "56side.jpg", height: 2448, width: 3264, description: "" },
    {
        name: "miata-2918-2919__82154_zoom.jpg",
        height: 480,
        width: 640,
        description: ""
    },
    {
        name: "BlueSteelInterior.jpg",
        height: 1380,
        width: 2082,
        description: ""
    },
    { name: "yy8390mustang.jpg", height: 304, width: 478, description: "" },
    { name: "56seats.jpg", height: 2448, width: 3264, description: "" },
    { name: "skiboat.jpg", height: 240, width: 440, description: "" },
    {
        name: "autointeriorline-art.jpg",
        height: 398,
        width: 406,
        description: ""
    },
    { name: "Bugeyetrunk2.jpg", height: 456, width: 755, description: "" },
    { name: "red356cab.jpg", height: 1200, width: 1600, description: "" },
    { name: "modelaorange.jpg", height: 1152, width: 1728, description: "" },
    { name: "BlackBlueHarley.jpg", height: 890, width: 1177, description: "" },
    { name: "Bugeyeinteriorjpg.jpg", height: 451, width: 744, description: "" },
    { name: "BillmodelA.jpg", height: 1061, width: 1394, description: "" },
    { name: "bmwinterior.jpg", height: 531, width: 640, description: "" },
    { name: "rearcorner.jpg", height: 361, width: 632, description: "" },
    { name: "IMG_0919.jpg", height: 374, width: 640, description: "" },
    { name: "porscheturbotiny.jpg", height: 947, width: 1184, description: "" },
    { name: "Rollsroyce81.jpg", height: 1637, width: 1456, description: "" },
    { name: "tomotis32.jpg", height: 819, width: 1126, description: "" },
    {
        name: "Kes58forwebassorted.jpg",
        height: 901,
        width: 1332,
        description: ""
    },
    { name: "BillsMerc.jpg", height: 1870, width: 1820, description: "" },
    { name: "bottlesview.jpg", height: 1075, width: 1613, description: "" },
    { name: "dashview.jpg", height: 1075, width: 1613, description: "" },
    { name: "sprinter2.jpg", height: 1152, width: 1728, description: "" },
    {
        name: "truckseatvinsert.jpg",
        height: 1936,
        width: 2592,
        description: ""
    },
    { name: "seat.jpg", height: 1728, width: 1152, description: "" },
    {
        name: "yellowspeedsterint.jpg",
        height: 322,
        width: 500,
        description: ""
    },
    { name: "Nelsonbench.jpg", height: 1152, width: 1728, description: "" },
    {
        name: "robbins_MiataGlasszipperless_2.jpg",
        height: 271,
        width: 439,
        description: ""
    },
    { name: "ambr1993.jpg", height: 450, width: 600, description: "" },
    {
        name: "CharlieSheen69Fury.jpg",
        height: 923,
        width: 1219,
        description: ""
    },
    { name: "56dash.jpg", height: 2448, width: 3264, description: "" },
    { name: "69Novainterior.jpg", height: 1152, width: 1728, description: "" },
    { name: "benz.jpg", height: 292, width: 500, description: "" },
    {
        name: "LeatherHandstitchededgesVWwhiteseats.jpg",
        height: 384,
        width: 576,
        description: ""
    },
    { name: "ctypedash.jpg", height: 1936, width: 2592, description: "" },
    {
        name: "Ferrariinterio2r.jpg",
        height: 2240,
        width: 3000,
        description: ""
    },
    { name: "IMG_7476.jpg", height: 746, width: 1728, description: "" },
    { name: "woodyfront_3332.jpg", height: 1936, width: 2592, description: "" },
    { name: "Kes58forwebBIG.jpg", height: 584, width: 1285, description: "" },
    { name: "IMG_7475.jpg", height: 864, width: 1728, description: "" },
    { name: "62RRbackseatjpg.jpg", height: 598, width: 800, description: "" },
    { name: "satchsseat_2980.jpg", height: 1936, width: 2592, description: "" },
    { name: "underdash.jpg", height: 1152, width: 1728, description: "" },
    { name: "rodsnwheels3rd.jpg", height: 326, width: 851, description: "" },
    { name: "MustangGT500Conv.jpg", height: 627, width: 2082, description: "" },
    { name: "satchs_2981.jpg", height: 2448, width: 3264, description: "" },
    { name: "IMG_4018.jpg", height: 1152, width: 1728, description: "" },
    { name: "ctypecar.jpg", height: 1936, width: 2592, description: "" },
    { name: "ctypeseats.jpg", height: 1936, width: 2592, description: "" },
    { name: "IMG_2095.jpg", height: 1936, width: 2592, description: "" },
    { name: "BM32Int.jpg", height: 1464, width: 2076, description: "" },
    { name: "nissan350Z.jpg", height: 249, width: 383, description: "" },
    { name: "IMG_3049.jpg", height: 1936, width: 2592, description: "" },
    { name: "RobbinsTopBanner.jpg", height: 365, width: 1178, description: "" },
    { name: "aprillalowres.jpg", height: 2056, width: 3861, description: "" },
    { name: "Panteraseats.jpg", height: 2592, width: 1936, description: "" },
    { name: "blackiesseats.jpg", height: 1152, width: 1728, description: "" },
    {
        name: "Porscheturbodash_4637.jpg",
        height: 1152,
        width: 1728,
        description: ""
    },
    {
        name: "porscheturbodoors_4634.jpg",
        height: 1152,
        width: 1728,
        description: ""
    },
    { name: "Ferrariinterior.jpg", height: 2241, width: 3000, description: "" }
];

const imagePaths = images.map(image => ({
    ...image,
    isLoading: false,
    isCached: false,
    // eslint-disable-next-line
    src: require(`../../photos/${image.name}`)
}));

export const imageState = imagePaths.reduce((acc, image) => {
    acc[image.name] = image;
    return acc;
}, {});
export default imagePaths;
