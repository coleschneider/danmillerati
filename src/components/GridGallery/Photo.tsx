import React from "react";
import { useInView } from "../../hooks/useInView";
import { useGalleryContext } from "../../pages/Gallery/Gallery";

const imgWithClick = { cursor: "pointer" };
// eslint-disable-next-line
interface Props {
    index: number;
    onClick: (
        e: React.SyntheticEvent,
        photo: GalleryImage & { index: number }
    ) => void;
    photo: GalleryImage;
    margin: number;
    top: number;
    left: number;
    direction: "row" | "column";
}
const Photo = ({
    index,
    onClick,
    photo,
    margin,
    direction,
    top,
    left
}: Props) => {
    const { getImageStatus, loadImage, imageLoading } = useGalleryContext();
    const [ref, inView, entry] = useInView({
        /* Optional options */
        triggerOnce: true,
        threshold: 0
    });

    const imgStyle = { margin, display: "block" } as React.CSSProperties;
    if (direction === "column") {
        imgStyle.position = "absolute";
        imgStyle.left = left;
        imgStyle.top = top;
    }

    const isLoaded = getImageStatus(photo.name);
    const imageStyle = onClick ? { ...imgStyle, ...imgWithClick } : imgStyle;
    const finalStyle = {
        ...imageStyle,
        // eslint-disable-next-line
        height: photo.height,
        // eslint-disable-next-line
        width: photo.width
    };

    React.useEffect(() => {
        if (inView) {
            loadImage(photo.name);
        }
        // eslint-disable-next-line
    }, [inView]);

    return (
        <div
            role="article"
            data-name={photo.name}
            className="img_lzy"
            ref={ref}
            onClick={e => onClick(e, { ...photo, index })}
            style={{
                ...finalStyle
            }}
        >
            {inView && (
                <img
                    alt=""
                    src={photo.src}
                    onLoad={() => {
                        imageLoading(photo.name);
                    }}
                    style={{
                        ...finalStyle,
                        opacity: isLoaded ? 1 : 0,
                        transition: "all 1s cubic-bezier(0.23, 1, 0.32, 1) 0s"
                    }}
                />
            )}
        </div>
    );
};

export default Photo;
