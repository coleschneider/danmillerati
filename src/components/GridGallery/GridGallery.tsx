/* eslint-disable */

import React, { useRef } from "react";
import useResizeObserver from "../../hooks/useResizeObserver";
import Photo from "./Photo";
import { computeColumnLayout } from "./layouts/columns";
import { computeRowLayout } from "./layouts/justified";
import { findIdealNodeSearch } from "../../utils/findAnchor";
import imagePaths, { imageState } from "../../pages/Gallery/imagePaths";
import { useGalleryContext } from "../../pages/Gallery/Gallery";
import Modal from "../Modal/Modal";

function GridGallery({    
    photos,    
    direction,    
    margin,    
    limitNodeSearch,    
    targetRowHeight,    
    columns,
    renderImage,
}:any) {
    const { carousel, goPrevImage, imageStatus, goNextImage,getImageStatus, setImage, openLightbox, imageLoading, loadImage, lightbox } = useGalleryContext();

    const galleryEl = useRef(null);
    const { containerWidth } = useResizeObserver({
        initialWidth: 0,
        container: galleryEl
    });
    const handleClick = (event, { index, name }) => {
        const hasLoaded =imageStatus[name]
        if(hasLoaded){
            setImage(index)
            openLightbox()
        }
        // onClick(event, {
        //     index,
        //     photo: photos[index],
        //     previous: photos[index - 1] || null,
        //     next: photos[index + 1] || null
        // });
    };

    // no containerWidth until after first render with refs, skip calculations and render nothing
    if (!containerWidth) return <div ref={galleryEl}>&nbsp;</div>;
    // subtract 1 pixel because the browser may round up a pixel
    const width = containerWidth - 1;
    let galleryStyle;
    let thumbs;

    if (direction === "row") {
        // allow user to calculate limitNodeSearch from containerWidth
        if (typeof limitNodeSearch === "function") {
            // eslint-disable-next-line
            limitNodeSearch = limitNodeSearch(containerWidth);
        }
        if (typeof targetRowHeight === "function") {
            // eslint-disable-next-line
            targetRowHeight = targetRowHeight(containerWidth);
        }
        // set how many neighboring nodes the graph will visit
        if (limitNodeSearch === undefined) {
            
            // eslint-disable-next-line
            limitNodeSearch = 2;
            if (containerWidth >= 450) {
                // eslint-disable-next-line
                limitNodeSearch = findIdealNodeSearch({
                    containerWidth,
                    targetRowHeight
                });
            }
        }

        galleryStyle = {
            display: "flex",
            flexWrap: "wrap",
            flexDirection: "row"
        };
        thumbs = computeRowLayout({
            containerWidth: width,
            limitNodeSearch,
            targetRowHeight,
            margin,
            photos
        });
    }
    if (direction === "column") {
        // allow user to calculate columns from containerWidth
        if (typeof columns === "function") {
            // eslint-disable-next-line
            columns = columns(containerWidth);
        }
        // set default breakpoints if user doesn't specify columns prop
        if (columns === undefined) {
            // eslint-disable-next-line
            columns = 1;
            // eslint-disable-next-line
            if (containerWidth >= 500) columns = 2;
            // eslint-disable-next-line
            if (containerWidth >= 900) columns = 3;
            // eslint-disable-next-line
            if (containerWidth >= 1500) columns = 4;
        }
        galleryStyle = { position: "relative" };
        thumbs = computeColumnLayout({
            containerWidth: width,
            columns,
            margin,
            photos
        });
        galleryStyle.height = thumbs[thumbs.length - 1].containerHeight;
    }

    const renderComponent = renderImage || Photo;
    const activeImage = imagePaths[carousel]

    return (
        <div className="react-photo-gallery--gallery">
            <div ref={galleryEl} style={galleryStyle}>
                {thumbs.map((thumb, index) => {
                    const { left, top, containerHeight, ...photo } = thumb;
                    return <Photo
                        left={left}
                        top={top}
                        key={thumb.key || thumb.src}
                        containerHeight={containerHeight}
                        index={index}
                        margin={margin}
                        direction={direction}
                        onClick={handleClick}
                        photo={photo}
                    
                    />
                })}

            </div>
        </div>
    );
}
const Gallery = React.memo(GridGallery);

// Gallery.propTypes = {
//   photos: PropTypes.arrayOf(photoPropType).isRequired,
//   direction: PropTypes.string,
//   onClick: PropTypes.func,
//   columns: PropTypes.oneOfType([PropTypes.func, PropTypes.number]),
//   targetRowHeight: PropTypes.oneOfType([PropTypes.func, PropTypes.number]),
//   limitNodeSearch: PropTypes.oneOfType([PropTypes.func, PropTypes.number]),
//   margin: PropTypes.number,
//   renderImage: PropTypes.func,
// };
// @ts-ignore
Gallery.defaultProps = {
    margin: 2,
    direction: "row",
    targetRowHeight: 300
};
export { Photo };
export default Gallery;
