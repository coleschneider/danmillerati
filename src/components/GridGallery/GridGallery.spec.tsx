import * as React from "react";
import { mount } from "enzyme";
import GridGallery from "./GridGallery";
import imagePaths from "../../pages/Gallery/imagePaths";
import createGlobalState from "../../hooks/useGlobalState";
import useGallery from "../../hooks/useGallery";
import * as useInView from "../../hooks/useInView/useInView";

jest.mock("../../pages/Gallery/imagePaths");
jest.mock("../../hooks/useInView/useInView");
describe("GridGallery", () => {
    let Wrapper: any;

    it("should work", () => {
        const galleryContext = createGlobalState(useGallery);
        const GalleryWrapper = () => {
            const context = React.useContext(galleryContext.Context);

            return <GridGallery photos={imagePaths} />;
        };
        const spy = jest
            .spyOn(useInView, "default")
            .mockImplementation(() => [null, true, null]);
        Wrapper = mount(
            <galleryContext.Provider>
                <GalleryWrapper />
            </galleryContext.Provider>
        );

        expect(Wrapper).toMatchSnapshot();
    });
});
