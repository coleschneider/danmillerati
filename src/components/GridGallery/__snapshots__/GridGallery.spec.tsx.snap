// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridGallery should work 1`] = `
<useGallery.Provider>
  <GalleryWrapper>
    <GridGallery
      direction="row"
      margin={2}
      photos={
        Array [
          Object {
            "description": "",
            "height": 190,
            "isCached": true,
            "isLoading": false,
            "name": "Backtobacklounge.jpg",
            "src": "Backtobacklounge.jpg",
            "width": 190,
          },
          Object {
            "description": "",
            "height": 190,
            "isCached": true,
            "isLoading": false,
            "name": "electricpontoon.jpg",
            "src": "electricpontoon.jpg",
            "width": 190,
          },
          Object {
            "description": "",
            "height": 384,
            "isCached": false,
            "isLoading": false,
            "name": "32roadsterred.jpg",
            "src": "32roadsterred.jpg",
            "width": 576,
          },
          Object {
            "description": "",
            "height": 216,
            "isCached": false,
            "isLoading": false,
            "name": "356Ccabdoor.jpg",
            "src": "356Ccabdoor.jpg",
            "width": 288,
          },
        ]
      }
      targetRowHeight={300}
    >
      <div>
        Â 
      </div>
    </GridGallery>
  </GalleryWrapper>
</useGallery.Provider>
`;
